$NetBSD: patch-aa,v 1.24 2011/09/25 14:33:09 schmonz Exp $

PKG_SYSCONFDIR and PKGMANDIR support, avoid GNUisms in standard utilities.

--- Makefile.PL.orig	2011-08-24 21:14:48.000000000 +0000
+++ Makefile.PL
@@ -19,11 +19,14 @@ SED?=sed
 
 # Additional configurable path variables.
 W3M_CGI_BIN?=$(PREFIX)/lib/w3m/cgi-bin
+SYSCONFDIR?=/etc/ikiwiki
+MANDIR?=$(PREFIX)/share/man
 
 tflag=$(shell if [ -n "$$NOTAINT" ] && [ "$$NOTAINT" != 1 ]; then printf -- "-T"; fi)
 extramodules=$(shell if [ "$$PROFILE" = 1 ]; then printf -- "-d:NYTProf"; fi)
 outprogs=ikiwiki.out ikiwiki-transition.out ikiwiki-calendar.out
 scripts=ikiwiki-update-wikilist ikiwiki-makerepo
+sysconfdir_scripts=ikiwiki-mass-rebuild ikiwiki-update-wikilist
 
 PROBABLE_INST_LIB=$(shell \\
 	if [ "$(INSTALLDIRS)" = "perl" ]; then \\
@@ -42,7 +45,7 @@ PROBABLE_INST_LIB=$(shell \\
 ikiwiki.setup:
 	HOME=/home/me $(PERL) -Iblib/lib $(extramodules) $(tflag) ikiwiki.in -dumpsetup ikiwiki.setup
 
-extra_build: $(outprogs) ikiwiki.setup docwiki
+extra_build: $(outprogs) ikiwiki.setup docwiki sysconfdir
 	./mdwn2man ikiwiki 1 doc/usage.mdwn > ikiwiki.man
 	./mdwn2man ikiwiki-mass-rebuild 8 doc/ikiwiki-mass-rebuild.mdwn > ikiwiki-mass-rebuild.man
 	./mdwn2man ikiwiki-makerepo 1 doc/ikiwiki-makerepo.mdwn > ikiwiki-makerepo.man
@@ -50,12 +53,15 @@ extra_build: $(outprogs) ikiwiki.setup d
 	./mdwn2man ikiwiki-update-wikilist 1 doc/ikiwiki-update-wikilist.mdwn > ikiwiki-update-wikilist.man
 	./mdwn2man ikiwiki-calendar 1 doc/ikiwiki-calendar.mdwn > ikiwiki-calendar.man
 	$(MAKE) -C po
-	$(SED) -i.bkp "s/Version:.*/Version: $(VER)/" ikiwiki.spec
+	$(PERL) -pi.bkp -e "s/Version:.*/Version: $(VER)/" ikiwiki.spec
 	rm -f ikiwiki.spec.bkp
 	
 docwiki:
 	$(PERL) -Iblib/lib $(extramodules) $(tflag) ikiwiki.in -setup docwiki.setup -refresh
 
+sysconfdir:
+	$(PERL) -pi -e "s|/etc/ikiwiki|$(SYSCONFDIR)|g" $(sysconfdir_scripts)
+	
 extra_clean:
 	$(PERL) -Iblib/lib $(extramodules) $(tflag) ikiwiki.in -setup docwiki.setup -clean
 	rm -f *.man $(outprogs) ikiwiki.setup plugins/*.pyc
@@ -69,8 +75,8 @@ underlay_install:
 	install -d $(DESTDIR)$(PREFIX)/share/ikiwiki
 	for dir in `cd underlays && $(FIND) . -follow -type d`; do \
 		install -d $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir; \
-		for file in `$(FIND) underlays/$$dir -follow -maxdepth 1 -type f -not -name \\*.full.js -not -name \\*.full.css`; do \
-			cp -aL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir 2>/dev/null || \
+		for file in `$(FIND) underlays/$$dir -follow -maxdepth 1 -type f ! -name \\*.full.js ! -name \\*.full.css`; do \
+			cp -pRL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir 2>/dev/null || \
 			install -m 644 $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir; \
 		done; \
 	done
@@ -79,7 +85,7 @@ underlay_install:
 	install -d $(DESTDIR)$(PREFIX)/share/ikiwiki/directives/ikiwiki/directive
 	for file in doc/ikiwiki/directive/*; do \
 		if [ -f "$$file" ]; then \
-			cp -aL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/directives/ikiwiki/directive 2>/dev/null || \
+			cp -pRL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/directives/ikiwiki/directive 2>/dev/null || \
 			install -m 644 $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/directives/ikiwiki/directive; \
 		fi \
 	done
@@ -94,7 +100,7 @@ underlay_install:
 			elif echo "$$file" | grep -q base.css; then \
 				:; \
 			elif [ -f "$$file" ]; then \
-				cp -aL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$file 2>/dev/null || \
+				cp -pRL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$file 2>/dev/null || \
 				install -m 644 $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$file; \
 			fi \
 		done; \
@@ -106,7 +112,7 @@ extra_install: underlay_install
 		install -d $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$dir; \
 	done
 	for file in `cd doc/examples; $(FIND) . -type f ! -regex '.*discussion.*'`; do \
-		cp -aL doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file 2>/dev/null || \
+		cp -pRL doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file 2>/dev/null || \
 		install -m 644 doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file; \
 	done
 
@@ -125,15 +131,15 @@ extra_install: underlay_install
 		install -m 755 $$file $(DESTDIR)$(PREFIX)/lib/ikiwiki/plugins; \
 	done
 
-	install -d $(DESTDIR)$(PREFIX)/share/man/man1
-	install -m 644 ikiwiki.man $(DESTDIR)$(PREFIX)/share/man/man1/ikiwiki.1
-	install -m 644 ikiwiki-makerepo.man $(DESTDIR)$(PREFIX)/share/man/man1/ikiwiki-makerepo.1
-	install -m 644 ikiwiki-transition.man $(DESTDIR)$(PREFIX)/share/man/man1/ikiwiki-transition.1
-	install -m 644 ikiwiki-update-wikilist.man $(DESTDIR)$(PREFIX)/share/man/man1/ikiwiki-update-wikilist.1
-	install -m 644 ikiwiki-calendar.man $(DESTDIR)$(PREFIX)/share/man/man1/ikiwiki-calendar.1
+	install -d $(DESTDIR)$(MANDIR)/man1
+	install -m 644 ikiwiki.man $(DESTDIR)$(MANDIR)/man1/ikiwiki.1
+	install -m 644 ikiwiki-makerepo.man $(DESTDIR)$(MANDIR)/man1/ikiwiki-makerepo.1
+	install -m 644 ikiwiki-transition.man $(DESTDIR)$(MANDIR)/man1/ikiwiki-transition.1
+	install -m 644 ikiwiki-update-wikilist.man $(DESTDIR)$(MANDIR)/man1/ikiwiki-update-wikilist.1
+	install -m 644 ikiwiki-calendar.man $(DESTDIR)$(MANDIR)/man1/ikiwiki-calendar.1
 	
-	install -d $(DESTDIR)$(PREFIX)/share/man/man8
-	install -m 644 ikiwiki-mass-rebuild.man $(DESTDIR)$(PREFIX)/share/man/man8/ikiwiki-mass-rebuild.8
+	install -d $(DESTDIR)$(MANDIR)/man8
+	install -m 644 ikiwiki-mass-rebuild.man $(DESTDIR)$(MANDIR)/man8/ikiwiki-mass-rebuild.8
 	
 	install -d $(DESTDIR)$(PREFIX)/sbin
 	install ikiwiki-mass-rebuild $(DESTDIR)$(PREFIX)/sbin
@@ -150,10 +156,10 @@ extra_install: underlay_install
 	
 	# These might fail if a regular user is installing into a home
 	# directory.
-	-install -d $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 wikilist $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto.setup $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto-blog.setup $(DESTDIR)/etc/ikiwiki
+	-install -d $(DESTDIR)$(SYSCONFDIR)
+	-install -m 0644 wikilist $(DESTDIR)$(SYSCONFDIR)
+	-install -m 0644 auto.setup $(DESTDIR)$(SYSCONFDIR)
+	-install -m 0644 auto-blog.setup $(DESTDIR)$(SYSCONFDIR)
 }
 }
 
